src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -310,4 +311,17 @@ public static boolean isSharedIndexStateRequired(KNNEngine knnEngine, String mod
         }
         return JNIService.isSharedIndexStateRequired(indexAddr, knnEngine);
     }
+
+    /**
+     * Tell if it is binary index or not
+     *
+     * @param knnEngine knn engine associated with an index
+     * @param parameters parameters associated with an index
+     * @return true if it is binary index
+     */
+    public static boolean isBinaryIndex(KNNEngine knnEngine, Map<String, Object> parameters) {
```

Comment:
Cant we check a field if its binary? Issue is that someday we may move away from the index description and manually build the index so I hate to add a dependence on it.

Reply:
Sure. Let me update the code.

Reply:
Updated.